python/lsst/cp/pipe/defects.py

```diff
@@ -0,0 +1,645 @@
+            msg = f"Found {len(defects)} defects containing {self._nPixFromDefects(defects)} pixels"
+            msg += " in master dark"
+            self.log.info(msg)
+            defectListFromDarks.append(defects)
+            if self.config.makePlots:
+                self._plot(dataRef, exp, visitList[0], self.config.nSigmaDark, defects, 'dark')
+
+            exp = dataRef.get('flat')
+            defects = self.findHotAndColdPixels(exp, 'flat')
+            msg = f"Found {len(defects)} defects containing {self._nPixFromDefects(defects)} pixels"
+            msg += " in master flat"
+            self.log.info(msg)
+            defectListFromFlats.append(defects)
+            if self.config.makePlots:
+                self._plot(dataRef, exp, visitList[0], self.config.nSigmaBright, defects, 'flat')
+
+        elif self.config.mode == 'VISITS':
+            butler = dataRef.getButler()
+
+            if self.config.assertSameRun:
+                runs = self._getRunListFromVisits(butler, visitList)
+                if len(runs) != 1:
+                    raise RuntimeError(f'Got data from runs {runs} with assertSameRun==True')
+
+            for visit in visitList:
+                imageType = butler.queryMetadata('raw', self.config.imageTypeKey, dataId={'visit': visit})[0]
+                imageType = imageType.lower()
+                dataRef.dataId['visit'] = visit
+
+                if imageType == 'flat':  # note different isr tasks
+                    exp = self.isrForFlats.runDataRef(dataRef).exposure
+                    defects = self.findHotAndColdPixels(exp, imageType)
+                    defectListFromFlats.append(defects)
+
+                elif imageType == 'dark':
+                    exp = self.isrForDarks.runDataRef(dataRef).exposure
+                    defects = self.findHotAndColdPixels(exp, imageType)
+                    defectListFromDarks.append(defects)
+
+                else:
+                    raise RuntimeError("Failed on imageType {imageType}. Only flats and darks supported")
+
+                msg = f"Found {len(defects)} defects containing {self._nPixFromDefects(defects)} pixels"
+                msg += f" in visit {visit}"
+                self.log.info(msg)
+
+                if self.config.makePlots:
+                    nSig = self.config.nSigmaBright if imageType == 'flat' else self.config.nSigmaDark
+                    self._plot(dataRef, exp, visit, nSig, defects, imageType)
+
+        self.log.info(f'Combining {len(defectListFromDarks)} defect sets from darks for detector {detNum}')
+        mergedDefectsFromDarks = self._postProcessDefectSets(defectListFromDarks, exp.getDimensions())
+        self.log.info(f'Combining {len(defectListFromFlats)} defect sets from flats for detector {detNum}')
+        mergedDefectsFromFlats = self._postProcessDefectSets(defectListFromFlats, exp.getDimensions())
+
+        self.log.info(f'Combining bright and dark defect sets')
```

Comment:
It's one of the processing steps, so I think it's at right level (but then again, I would, because I put it like that, so I'm probably not the best judge...)